crates/egui_demo_app/src/apps/image_viewer.rs

```diff
@@ -52,28 +53,31 @@ impl Default for ImageViewer {
 
 impl eframe::App for ImageViewer {
     fn update(&mut self, ctx: &egui::Context, _: &mut eframe::Frame) {
-        egui::TopBottomPanel::new(TopBottomSide::Top, "url bar").show(ctx, |ui| {
-            ui.horizontal_centered(|ui| {
-                let label = ui.label("URI:");
-                ui.text_edit_singleline(&mut self.uri_edit_text)
-                    .labelled_by(label.id);
-                if ui.small_button("✔").clicked() {
-                    ctx.forget_image(&self.current_uri);
-                    self.uri_edit_text = self.uri_edit_text.trim().to_owned();
-                    self.current_uri = self.uri_edit_text.clone();
-                };
-
-                #[cfg(not(target_arch = "wasm32"))]
-                if ui.button("file…").clicked() {
-                    if let Some(path) = rfd::FileDialog::new().pick_file() {
-                        self.uri_edit_text = format!("file://{}", path.display());
+        egui::Panel::new(PanelSide::Horizontal(HorizontalSide::Top), "url bar").show(
```

Comment:
We should use the helpers instead

```suggestion
        egui::Panel::left("url bar").show(
```